app/src/main/java/io/apicurio/registry/mt/TenantContextImpl.java

```diff
@@ -16,39 +16,43 @@
 
 package io.apicurio.registry.mt;
 
-import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.RequestScoped;
 
 /**
  * @author eric.wittmann@gmail.com
  */
-@ApplicationScoped
+@RequestScoped
```

Comment:
I didn't notice this change at first, which made the rest of the changes in this class INSANE.  :)